### DISCUSSION TITLE
Compile Issue

### DISCUSSION THREAD

**Discussion by DirkLewis** [drseg/swift-fsm#4] [General]
So I got up to continue playing with FSM and I started getting "Command SwiftCompile failed with a nonzero exit code".  After digging around performing clean up of Xcode 16, reinstalling packages, starting a new project etc. The only thing that would let me build was to comment all of the places where I called define(state){ when | then | method) statements.  Once removed I could build, put back and Command SwiftCompile failed with a nonzero exit code again. 

any insights?

**Comment by drseg** [comment]
I apologise for not seeing these, somehow I didn't have any notifications on for this section and missed it.

First, if you are still looking into this, are you still encountering the problem with 0.9.4? I use both the development and release versions of this in some personal projects of mine and haven't seen problems like this for quite a while. There were some issues of this kind in the early days though.

Regarding your other question about 'getting the state', indeed there is no way to get the state. In my opinion it would defeat the purpose of a state machine, because if you are asking it what its state is, then by definition you are creating another state machine outside it that depends on querying its state. This would seem to be encouraging a mess, and so it was intentionally disallowed, but if you can describe why you would want this, it would not be difficult to enable.

It seems perhaps you were looking to get hold of the state just to verify that it was working as expected? If you look at the internal tests, you'll see even those are mostly done by verifying the output chain, not the internal state. That's in part to make sure that it's the behaviour that's being verified and not an implementation that might change.
